### DISCUSSION TITLE
如何设置默认窗口Size？

### DISCUSSION THREAD

**Discussion by terrihonkai** [opensumi/core#2726] [Q&A] [answered]
目前用的ide-electron项目，发现窗口打开都是一个比较小的尺寸，通过哪里设置可以默认全屏呢？

**Comment by bytemain** [comment]
有一种方案是通过模块在 onStart 生命周期中执行命令将应用全屏。

参考下这个代码创建一个模块：https://github.com/opensumi/opensumi-module-samples/blob/main/modules/builtin-events/browser/builtin-events.contribution.ts#L15

然后在 onStart 函数里调用 electron 相关的 API 即可：
```ts
import { IElectronMainLifeCycleService, IElectronMainUIService } from '@opensumi/ide-core-common/lib/electron';

@Domain(ClientAppContribution)
export class XXXXContribution implements ClientAppContribution {

  @Autowired(IElectronMainLifeCycleService)
  private electronMainLifeCycleService: IElectronMainLifeCycleService;

  async onStart() {
    // 最大化
    this.electronMainLifeCycleService.maximizeWindow(electronEnv.currentWindowId);
    // 全屏
    // this.electronMainLifeCycleService.fullscreenWindow(electronEnv.currentWindowId);
  }
}
```

**Comment by bytemain** [comment, marked as answer]
上面的这个效果是打开窗口后再将窗口最大化，如果你需要实现更复杂的效果就需要自己把代码写在 Electron Main 层。目前这部分开源代码实现还比较简陋，需要开发者自己写相关逻辑。

这一行代码会启动窗口：https://github.com/opensumi/ide-electron/blob/main/src/main/launch.ts#L96
你可以在通过设置 `loadWorkspace(undefined, undefined, {show: false})` 来不显示窗口。
然后通过 electron 自己的 API 来设置该 browserWindow 的大小：
```ts
const codeWindow =  electronApp.loadWorkspace(undefined, { launchToOpenFile: workspace }, {show: false});
const browserWindow = window.getBrowserWindow();
// 然后就可以操作这个 window 了
browserWindow.maximize();
// browserWindow.setFullScreen(true);
browserWindow.show();
```
